tests/raftstore/cluster.rs

```diff
@@ -459,6 +460,46 @@ impl<T: Simulator> Cluster<T> {
         assert!(status_resp.has_region_detail());
         status_resp.take_region_detail()
     }
+
+    // NOTE: if you have set transport hook before, call this function will clear them
+    pub fn partition(&mut self, s1: Arc<HashSet<u64>>, s2: Arc<HashSet<u64>>) {
+        for node_id in s1.as_ref() {
+            let filter = new_partition_filter(s2.clone());
+            self.sim.rl().hook_transport(*node_id, vec![filter]);
+        }
+        for node_id in s2.as_ref() {
+            let filter = new_partition_filter(s1.clone());
+            self.sim.wl().hook_transport(*node_id, vec![filter]);
+        }
+    }
+
+    pub fn reset_transport_hooks(&mut self) {
+        let sim = &self.sim.rl();
+        for node_id in sim.get_node_ids() {
+            sim.hook_transport(node_id, vec![]);
+        }
+    }
+}
+
+struct PartitionFilter {
+    node_ids: Arc<HashSet<u64>>,
```

Comment:
suggested by @BusyJay 
this HashSet is read only, so we can share it between multi PartitionFilter, thus don't need to make multi HashSet copies